File: SDKs/plaid.com/2020-09-14_1.20.6/java/lib/src/main/java/openapisdk/models/shared/SandboxPublicTokenCreateRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * SandboxPublicTokenCreateRequest
 * SandboxPublicTokenCreateRequest defines the request schema for `/sandbox/public_token/create`
**/
public class SandboxPublicTokenCreateRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("client_id")
    public String clientId;
    public SandboxPublicTokenCreateRequest withClientId(String clientId) {
        this.clientId = clientId;
        return this;
    }
    @JsonProperty("initial_products")
    public ProductsEnum[] initialProducts;
    public SandboxPublicTokenCreateRequest withInitialProducts(ProductsEnum[] initialProducts) {
        this.initialProducts = initialProducts;
        return this;
    }
    @JsonProperty("institution_id")
    public String institutionId;
    public SandboxPublicTokenCreateRequest withInstitutionId(String institutionId) {
        this.institutionId = institutionId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("options")
    public SandboxPublicTokenCreateRequestOptions options;
    public SandboxPublicTokenCreateRequest withOptions(SandboxPublicTokenCreateRequestOptions options) {
        this.options = options;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("secret")
    public String secret;
    public SandboxPublicTokenCreateRequest withSecret(String secret) {
        this.secret = secret;
        return this;
    }
}

File: SDKs/plaid.com/2020-09-14_1.20.6/python/src/sdk/models/shared/sandboxpublictokencreaterequest.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import products_enum as shared_products_enum
from ..shared import sandboxpublictokencreaterequestoptions as shared_sandboxpublictokencreaterequestoptions


@dataclass_json
@dataclasses.dataclass
class SandboxPublicTokenCreateRequest:
    r"""SandboxPublicTokenCreateRequest
    SandboxPublicTokenCreateRequest defines the request schema for `/sandbox/public_token/create`
    """
    
    initial_products: list[shared_products_enum.ProductsEnum] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('initial_products') }})
    institution_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('institution_id') }})
    client_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('client_id') }})
    options: Optional[shared_sandboxpublictokencreaterequestoptions.SandboxPublicTokenCreateRequestOptions] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('options') }})
    secret: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('secret') }})
    
